packages/cadl-rlc-emitter/test/unit/generate/models.spec.ts

```diff
@@ -0,0 +1,452 @@
+      prop: string;
+    }
+    `);
+    assert.ok(schemaOutput);
+    const { inputModelFile, outputModelFile } = schemaOutput!;
+    // both are undefined
+    assert.ok(!inputModelFile);
+    assert.ok(!outputModelFile);
+  });
+
+  async function verifyPropertyType(
+    cadlType: string,
+    inputType: string,
+    options?: VerifyPropertyConfig
+  ) {
+    const defaultOption: VerifyPropertyConfig = {
+      additionalCadlDefinition: "",
+      outputType: inputType,
+      additionalInputContent: "",
+      additionalOutputContent: ""
+    };
+    const {
+      additionalCadlDefinition,
+      outputType,
+      additionalInputContent,
+      additionalOutputContent
+    } = {
+      ...defaultOption,
+      ...options
+    };
+    const schemaOutput = await generateInputOutputModelsFromCadl(`
+    ${additionalCadlDefinition}
+    model InputOutputModel {
+      prop: ${cadlType};
+    }
+    @route("/models")
+    @get
+    op getModel(@body input: InputOutputModel): InputOutputModel;`);
+    assert.ok(schemaOutput);
+    const { inputModelFile, outputModelFile } = schemaOutput!;
+    assert.strictEqual(inputModelFile?.path, "models.ts");
+    assertEqualContent(
+      inputModelFile?.content!,
+      `
+    export interface InputOutputModel {
+        prop: ${inputType};
+    }
+    ${additionalInputContent}`
+    );
+
+    assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+    assertEqualContent(
+      outputModelFile?.content!,
+      `
+    export interface InputOutputModelOutput {
+      prop: ${outputType};
+    }
+    ${additionalOutputContent}`
+    );
+  }
+
+  describe("number generation", () => {
+    it("should handle int32 -> number", async () => {
+      await verifyPropertyType("int32", "number");
+    });
+    it("should handle int64 -> number", async () => {
+      await verifyPropertyType("int64", "number");
+    });
+    it("should handle safeint -> number", async () => {
+      await verifyPropertyType("safeint", "number");
+    });
+    it("should handle float32 -> number", async () => {
+      await verifyPropertyType("float32", "number");
+    });
+    it("should handle float64 -> number", async () => {
+      await verifyPropertyType("float64", "number");
+    });
+  });
+  describe("string generation", () => {
+    // TODO: how to handle extensible enum is not fanilized this could be changed in future
+    // issue tracked https://github.com/Azure/autorest.typescript/issues/1524
+    xit("should handle extensible_enum -> string", async () => {
+      // When extensible_enum is comsumed as body property it should be string only
+      const cadlTypeDefinition = `
+      @knownValues(TranslationLanguageValues)
+      model TranslationLanguage is string;
+      enum TranslationLanguageValues {
+        English,
+        Chinese,
+      }`;
+      const cadlType = "TranslationLanguage";
+      // TODO: ensure this trick is working
+      const typeScriptType = `"English" | "Chinese" | string & {}`;
```

Comment:
I don't think this is decided, it's just a proposal still needs some discussion about it.  Currently, we are still using string for extensible enum. @joheredi  right ?

Reply:
Yes I just ignore this case because it needs effort to finalize the current design. 

Generally I ignore below two cases 

- The behavior is unexpected 1) maybe incorrect output 2) error throw in our code ==> We need to fix our code to pass them
- The behavior is un-decided ==> We need to take time to re-think how to design better